File: prof.py
import time

from xipc import hack

with hack("py_called_1", "py_called_2", "js_called_1", "js_called_2") as mods:
    now = time.time()
    b = 0
    for x in range(1000):
        b += mods.py_called_1.add_ints(1, 2)  # type: ignore
    print("Time in seconds for 1000 additions using sockets:", time.time() - now)
    now = time.time()
    b = 0
    for x in range(1000):
        b += 1 + 2
    print("Time in seconds for 1000 additions:", time.time() - now)

File: prof.js
const { hack } = require("./xipc");
hack("py_called_1")(async (mods) => {
  let res1;
  let res2;
  const py_called_1 = mods.py_called_1;
  let now = new Date().getTime();
  let i = 0;
  for (; i < 1000; i++) {
    await py_called_1("add_ints")(1, 2);
  }
  console.log(
    "Time in seconds for 1000 additions using sockets:",
    (new Date().getTime() - now) / 1000
  );
  now = new Date().getTime();
  i = 0;
  for (; i < 1000; i++) {
    const q = 1 + 2;
  }
  console.log(
    "Time in seconds for 1000 additions:",
    (new Date().getTime() - now) / 1000
  );
});
